hal/src/rtl872x/i2c_hal.cpp

```diff
@@ -188,15 +210,17 @@ class I2cClass {
         conf.mode = OUTPUT_OPEN_DRAIN_PULLUP;
         conf.set_value = true;
         conf.value = 1;
-        hal_gpio_configure(sdaPin_, &conf, nullptr);
-        conf.value = hal_gpio_read(sclPin_);
         hal_gpio_configure(sclPin_, &conf, nullptr);
+        hal_gpio_configure(sdaPin_, &conf, nullptr);
+
+        // Check if slave is stretching the SCL
+        if (!WAIT_TIMED(transConfig_.timeout_ms, hal_gpio_read(sclPin_) == 0)) {
+            // We can't proceed because there is probably hardware issue.
+            SPARK_ASSERT(false);
```

Comment:
Return an error instead, this is not critical, the application or Device OS can potentially have some other means to resolve the issue with a problematic component.